tests/bugs/c-style-cast-overload.slang

```diff
@@ -0,0 +1,30 @@
+// https://github.com/shader-slang/slang/issues/2189
+// See also ./c-style-cast-coerce.slang which repsents the essence of this bug
```

Comment:
I somehow spelled "represents" even more incorrectly than the impulsive correction I made above...